### ISSUE TITLE
Types do not resolve in 3.11.0

### ISSUE THREAD

**Issue by dustinsoftware** [emberjs/data#6217]
### Reproduction

1. Clone https://github.com/dustinsoftware/ember-data-issue-repro/
1. `ember build`
1. Observe type error resolving `DS.Store`

### Description

Types are not resolving in `ember-data@3.11.0` with the types provided in `@types/ember-data`. I suspect an empty type definition is being shipped that is causing Typescript to ignore the types from DefinitelyTyped. cc @mike-north 

![image](https://user-images.githubusercontent.com/942358/60552146-b8a42400-9cfb-11e9-9721-572a0c621dc6.png)

### Versions

Run the following command and paste the output below: `npm ls ember-source && npm ls ember-cli && npm ls ember-data`.

```cli
e-data-ts@0.0.0 C:\code\e-data-ts
`-- ember-source@3.10.2

e-data-ts@0.0.0 C:\code\e-data-ts
`-- ember-cli@3.11.0-beta.3

e-data-ts@0.0.0 C:\code\e-data-ts
`-- ember-data@3.11.0
```

*P.S. If any of the packages show more than one installed version, that may be the root cause of the issue!*

--------------------------------------------------------------

Thanks again!

The `ember-data` Team <3

**Comment by richard-viney** [comment]
As mentioned on Discord, I've worked around this issue by adding the following to `package.json`:

```
  "scripts": {
    "postinstall": "rm -f node_modules/ember-data/stripped-types.d.ts"
  }
```

**Comment by mike-north** [comment]
@runspired looks like this mechanism is no longer going to work for us. We should switch to stripping all `.d.ts` files from packages in a prepublish step

**Comment by dustinsoftware** [comment]
Fixed in #6256

**Comment by runspired** [comment]
Ah yes, thanks @dustinsoftware forgot to close :)
